distributed/tests/test_client.py

```diff
@@ -331,6 +330,7 @@ def test_retries_get(c):
         x.compute()
 
 
+@pytest.mark.xfail(reason="Is this a sane thing to do?")
 @gen_cluster(client=True)
 async def test_compute_persisted_retries(c, s, a, b):
```

Comment:
This is a behavioral change. This test tests a condition where a persisted result is annotated by a subsequent compute call.

I don't believe something like this should be allowed since annotations have non-trivial effects on scheduling and we're very likely not handling all of these cases. For instance, this test is annotating a retry. The only reason why this works is because persist+compute are submitted basically in the same event loop tick such that the retry annotation is set before the task even has a chance to be computed.